tests/test_olio_grid_functions.py

```diff
@@ -0,0 +1,236 @@
+import os
+import numpy as np
+import pytest
+
+from resqpy.olio.grid_functions import left_right_foursome, infill_block_geometry, resequence_nexus_corp, random_cell, determine_corp_ijk_handedness, determine_corp_extent, translate_corp, triangles_for_cell_faces, actual_pillar_shape
+import resqpy.model as rq
+import resqpy.grid as grr
+
+
+def test_infill_block_geometry():
+
+    # --------- Arrange ----------
+    x_ = np.array([2, 4, 6, 8])
+    y_ = np.array([1, 3, 5, 7])
+    depth_ = np.array([10.0, 30.0, 50.0, 70.0])
+    x, y, depth = np.meshgrid(x_, y_, depth_)
+    extent = (4, 4, 4)
+    thickness = np.ones(extent) * 15
+
+    # Set two inactive cells
+    depth[1, 2, 1] = 0.001  # below the default depth_zero_tolerance so should be set to 0 by function
+    depth[1, 2, 2] = 0.001
+    # Assertions to check that depth does have 2 values == 0.001
+    assert len(depth[depth == 0.001]) == 2
+    assert np.unique(thickness) == np.array([15.])
+    # --------- Act ----------
+    infill_block_geometry(extent = extent, depth = depth.transpose(), thickness = thickness, x = x, y = y)
+    # --------- Assert ----------
+    assert len(depth[depth == 0]) == 0
+    assert len(thickness[thickness == 22.5]) == 2
```

Comment:
With snap_to_top_and_base defaulting to True, I don't think this is the answer we want here (though it might well be what the old code produced).  Let's discuss.